python_modules/dagster/dagster/daemon/sensor.py

```diff
@@ -184,39 +185,50 @@ def execute_sensor_iteration_loop(instance, workspace, logger, until=None):
     """
     workspace_loaded_time = pendulum.now("UTC").timestamp()
 
-    workspace_iteration = 0
-    start_time = pendulum.now("UTC").timestamp()
-    while True:
-        start_time = pendulum.now("UTC").timestamp()
-        if until and start_time >= until:
-            # provide a way of organically ending the loop to support test environment
-            break
-
-        if start_time - workspace_loaded_time > RELOAD_WORKSPACE:
-            workspace.cleanup()
-            workspace_loaded_time = pendulum.now("UTC").timestamp()
-            workspace_iteration = 0
+    max_workers = int(os.getenv("DAGSTER_SENSOR_DAEMON_THREADPOOL_WORKERS", "3"))
```

Comment:
Questions in my mind about this:
- do we want to make this opt in while its new? If it's opt-in, how should it be configured? Do we need a helm chart hook?
- 3 was not chosen here in the WIP with any rigor at all - how should we go about setting a default here?

I had been considering the idea of making it opt-in by using the sychronous executor that was there in the previous rev of this PR

Reply:
Yeah, I think adding instance settings might make sense here, but that probably means hooks in the helm chart.

The synchronous option as default makes sense here... 

I'd also consider keeping it asynchronous (with a default worker count of 1) and add a `wait_for_*` call at the end of the sensor iteration?  That might mitigate the concerns around falling behind, but you'd still mitigate the IO-bound gaps.

Reply:
Resolved in https://github.com/dagster-io/dagster/pull/8642...